nucypher/blockchain/eth/interfaces.py

```diff
@@ -143,10 +143,15 @@ def get_balance(self):
 
         @property
         def insufficient_eth(self) -> str:
-            gas = (self.payload.get('gas', 1) * self.payload['gasPrice'])  # FIXME: If gas is not included...
-            cost = gas + self.payload.get('value', 0)
-            message = f'{self.payload} from {self.payload["from"][:8]} - {self.base_message}.' \
-                      f'Calculated cost is {cost} but sender only has {self.get_balance()}.'
+            try:
+                transaction_fee = self.payload['gas'] * self.payload['gasPrice']
+            except KeyError:
+                return self.default
```

Comment:
I guess this commit never made it in because #2369  wasn't merged... @KPrasch cherry-picked it

Reply:
Yeah, and I actually changed it a bit more since then.